### PR TITLE
Add tests for concat with empty array

### PR DESCRIPTION
**Pull request by MatejUrbanQC**
The shape inference fails when inputs are eager. Added a xfail test case for it.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -634,3 +634,39 @@ def test_array_creation_with_invalid_fields():
 def test_promote_nullable():
     with pytest.warns(DeprecationWarning):
         assert ndx.promote_nullable(np.int64) == ndx.nint64
+
+
+@pytest.mark.xfail
```
Comment (adityagoel4512) [inline tests/test_core.py:639]:
```suggestion
@pytest.mark.xfail(reason="https://github.com/onnx/onnx/issues/6276")
```

**Review by adityagoel4512 — CHANGES_REQUESTED** [review]

**Review by cbourjau — APPROVED** [review]
